OpenRA.Mods.Common/AI/SupportPowerDecision.cs

```diff
@@ -160,7 +160,10 @@ public int GetAttractiveness(Actor a, Stance stance, Player firedBy)
 
 						case DecisionMetric.Health:
 							var health = a.TraitOrDefault<Health>();
-							return (health != null) ? (health.HP / health.MaxHP) * Attractiveness : 0;
+
+							// Cast to long to avoid overflow when multiplying by the health
+							var hp = (long)health.HP;
+							return (int)((health != null) ? (hp / health.MaxHP) * Attractiveness : 0);
```

Comment:
Can you please change this to
```csharp
if (health == null)
	return 0;

return (int)((long)health.HP * Attractiveness / health.MaxHP);
```

Otherwise you'll crash if `health == null`.  This also fixes an issue where it would return 0 unless the actor was at full health.

Reply:
Nice catch!